### PR TITLE
Refactor vagrant integration tests

### PR DESCRIPTION
**Pull request by lbogdan**
#### What type of PR is this?

/kind ci

#### What this PR does / why we need it:

This PR refactors the `vagrant` integration tests. It extracts the inline provisioning script (which installs the packages, bootstraps a Kubernetes cluster, and tests that it comes up successfully) from `Vagrantfile` to standalone script files.

There are separate scripts for installing the packages per distro (`install-deb.sh` and `install-rpm.sh`), but a single one (`test.sh`) for bootstrapping and testing the cluster.

> [!NOTE]
> [`lima` doesn't seem to support running external scripts on provisioning](https://github.com/lima-vm/lima/blob/master/examples/default.yaml), so the `lima.yaml` configs were not touched.

Later edit:

> [!NOTE]
> In order to be able to extract the Kubernetes version and `cri-o` package path to `versions.sh` (so we have them in one place) and source that from `install-*.sh`, we first copy all scripts to `/tmp/provision` inside the VM, and then we run them from there.

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

Related Slack thread: https://kubernetes.slack.com/archives/CAZH62UR1/p1717060130128699 .

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: test/scripts/install-deb.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
```
Comment (saschagrunert) [inline test/scripts/install-deb.sh:1]:
```suggestion
#!/usr/bin/env bash
```

File: test/scripts/install-rpm.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
```
Comment (saschagrunert) [inline test/scripts/install-rpm.sh:1]:
```suggestion
#!/usr/bin/env bash
```

File: test/scripts/install-deb.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+set -euxo pipefail
+
+# This script installs the required deb packages to bootstrap a Kubernetes cluster
+# It is referenced from ../deb/Vagrantfile
+
+# Package configs
+KUBERNETES_VERSION=v1.30
+PROJECT_PATH=prerelease:/main
```
Comment (saschagrunert) [inline test/scripts/install-deb.sh:10]: Do we want to source this from another file and include it here as well as in install-rpm.sh?

File: test/scripts/test.sh [hunk 4]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
```
Comment (saschagrunert) [inline test/scripts/test.sh:1]:
```suggestion
#!/usr/bin/env bash
```

File: test/scripts/install-deb.sh (hunk 3 above)
Comment (lbogdan) [reply test/scripts/install-deb.sh:10]: It would be nice, but I don't think `vagrant` tracks script dependencies.

**Review by saschagrunert — APPROVED** [review]
Thanks!

File: test/scripts/install-deb.sh (hunk 3 above)
Comment (lbogdan) [reply test/scripts/install-deb.sh]: Implemented in [commit](https://github.com/cri-o/packaging/pull/112/commits/831c9caa7dd23792d65a6275977921579b607a63).

**Comment by lbogdan** [comment]
@saschagrunert Sorry, I pushed a few more commits, mostly to make `shellcheck` happy, and be more explicit about it, this is now ready for a final review.

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**